numpy/core/einsumfunc.py

```diff
@@ -884,39 +886,68 @@ def einsum(*operands, **kwargs):
     -----
     .. versionadded:: 1.6.0
 
+    The Einstein summation convention can be used to compute
+    many multi-dimensional array operations. Additionally, the explicit
+    mode of this function provides further flexibility that can be used
+    to compute other array operations that might otherwise not be
+    considered classical Einstein summation operations. `einsum`
+    provides a succinct way of representing many array operations in a
+    clear and consistent format.
+
+    A non-exhaustive list of array operations which can be computed by
+    `einsum` is shown below along with examples,
+
+        * Trace of an array.
+        * Return a diagonal.
+        * Array axis summations.
+        * Transpositions and permutations.
+        * Matrix multiplication and vector dot product.
+        * Broadcasting and scalar multiplication.
+        * Tensor contractions.
+        * Chained array operations, in efficient calculation order.
+
     The subscripts string is a comma-separated list of subscript labels,
     where each label refers to a dimension of the corresponding operand.
-    Repeated subscripts labels in one operand take the diagonal.  For example,
-    ``np.einsum('ii', a)`` is equivalent to ``np.trace(a)``.
-
-    Whenever a label is repeated, it is summed, so ``np.einsum('i,i', a, b)``
-    is equivalent to ``np.inner(a,b)``.  If a label appears only once,
+    Whenever a label is repeated it is summed, so ``np.einsum('i,i', a, b)``
+    is equivalent to ``np.inner(a,b)``. If a label appears only once,
     it is not summed, so ``np.einsum('i', a)`` produces a view of ``a``
-    with no changes.
-
-    The order of labels in the output is by default alphabetical.  This
+    with no changes. A further example ``np.einsum('ij,jk', a, b)``
+    describes traditional matrix
+    multiplication and is equivalent to ``np.matmul(a,b)``. Repeated
```

Comment:
This is true only when a and b are 2d.

Reply:
Agreed but they are 2d by implication from the subscripts. Do you not consider it clearer for the reader to naturally assume 2d arrays in combination with matrix multiplication, rather than specifying the shape with additions to the description?
If your comment specifically regards `matmul` being capable of operating on arrays of other dimensions I would prefer to consider this out of scope for this function's docstring, since this example's purpose is only to demonstrate the familiar 'ij,jk' string format relative to matrix multiplication.

Alternatively I can just remove "and is equivalent to np.matmul(a,b)" but I do consider it to be of some value here?